### PR TITLE
Update to WordPressCS 2.0.0

### LINKED ISSUE #193: WPCS 2.0.0: upgrade sniffs
... and make other relevant changes.

Once WPCS 2.0.0 comes out (expected in about 2 to 3 weeks), some changes will be needed in the codebase.

I'll send in a PR once WPCS 2.0.0 has been released.

Refs:
* https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Upgrade-Guide-to-WordPressCS-2.0.0-for-Developers-of-external-standards
* https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1610

### PR DESCRIPTION
**Pull request by jrfnl**
[WordPressCS 2.0.0](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/releases/tag/2.0.0) has just been released and contains breaking changes.

This PR makes all the updates needed for WPTRT to move to a minimum WPCS requirement of 2.0.0.

Also see: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Upgrade-Guide-to-WordPressCS-2.0.0-for-Developers-of-external-standards

Fixes #193 

###  Composer: update requirements for WPCS 2.0.0

* Require WPCS 2.0.0.
* As WPCS 2.0.0 now has a minimum PHPCS requirement of PHPCS 3.3.1, the requirement for PHPCS should be in-line with that.

### PHPUnit: Remove test bootstrap

As WPCS 2.0.0 has dropped the `PHPCSAliases` file, there is now only one (external) bootstrap file which needs to be loaded to run the unit tests: the PHPCS native bootstrap which provided cross-version support for various PHPUnit versions.

With only one bootstrap file to load, we can just let PHPUnit handle this.

So, this commit removes the TRTCS  native `bootstrap.php` file and makes some related adjustments.

### Ruleset tweaks: remove `encoding`

The default `encoding` as of PHPCS 3.0.0 is `utf-8`, so we don't actually need to set this.

### Ruleset: add dev note for the future

### CODE REVIEW

**Comment by dingo-d** [comment]
You've beat me to it :smile: I'll take a look at the PR 🙂

File: composer.json [hunk 1]
```diff
@@ -52,11 +52,21 @@
...
 	"scripts"    : {
-		"install-standards"    : "@php ./vendor/squizlabs/php_codesniffer/bin/phpcs --config-set installed_paths ../../..,../../wp-coding-standards/wpcs,../../phpcompatibility/phpcompatibility-wp,../../phpcompatibility/phpcompatibility-paragonie",
-		"run-tests"            : "@php ./vendor/phpunit/phpunit/phpunit --filter WPThemeReview ./vendor/squizlabs/php_codesniffer/tests/AllTests.php",
-		"phpcs-i"              : "@php ./vendor/squizlabs/php_codesniffer/bin/phpcs -i",
-		"check-cs"             : "@php ./vendor/squizlabs/php_codesniffer/bin/phpcs",
-		"fix-cs"               : "@php ./vendor/squizlabs/php_codesniffer/bin/phpcbf"
+		"install-standards": [
+			"@php ./vendor/squizlabs/php_codesniffer/bin/phpcs --config-set installed_paths ../../..,../../wp-coding-standards/wpcs,../../phpcompatibility/phpcompatibility-wp,../../phpcompatibility/phpcompatibility-paragonie"
+		],
+		"run-tests": [
```
Comment (dingo-d) [inline composer.json]: If we revert this to just a string (since we are not executing multiple commands here), will this fix the Travis failure?

Comment (jrfnl) [reply composer.json]: No, it won't.

I've just had a look and most of the Travis failure was just Travis being weird, the rest of it was me forgetting to update the minimum supported PHPCS version in the Travis script. Oops :flushed: 

I've fixed up the initial Composer update commit now to include the change to the Travis script and I expect the build will now pass.

**Comment by jrfnl** [comment]
> You've beat me to it

Well, I did mention in #193 that I would send in a PR for this. I'd opened the issue to prevent people doing double work as I had the branch ready already.

**Comment by dingo-d** [comment]
All seems to be passing, I'll merge this to develop and ask people to test before releasing an update. That sounds ok?

**Comment by jrfnl** [comment]
@dingo-d Sounds good to me.

**Pull request merged**